### ISSUE TITLE
Rolling back snapshot prints grub-install: error: cannot find EFI directory.

### ISSUE THREAD

**Issue by Nightliss** [hrotkogabor/manjaro-btrfs#4]
Hi,

I've been testing these wonderful scripts on a manjaro 21.0 gnome in a virtualbox and came upon this error when restoring snapshot with `sudo btrfs_snapper_do_rollback.sh`

```
sudo btrfs_snapper_do_rollback.sh
[sudo] password for nightliss: 
calling snapper rollback
Ambit is classic.
Creating read-only snapshot of default subvolume. (Snapshot 13.)
Creating read-write snapshot of current subvolume. (Snapshot 14.)
Setting default subvolume to snapshot 14.
updating and re-installing grub
Using /dev/sda3, and /dev/sda as parent.
mkdir: cannot create directory '/root/.config/manjaro-tools': Read-only file system
mkdir: cannot create directory '/root/.config/manjaro-tools': Read-only file system
Generating grub configuration file ...
Found linux image: /boot/vmlinuz-5.10-x86_64
Found initrd image: /boot/intel-ucode.img /boot/initramfs-5.10-x86_64.img
Found initrd fallback image: /boot/initramfs-5.10-x86_64-fallback.img
Warning: os-prober will be executed to detect other bootable partitions.
It's output will be used to detect bootable binaries on them and create new boot entries.
mkdir: cannot create directory ‘/var/lib/os-prober/mount’: No such file or directory
mkdir: cannot create directory ‘/var/lib/os-prober/mount’: No such file or directory
mkdir: cannot create directory ‘/var/lib/os-prober/mount’: No such file or directory
Adding boot menu entry for UEFI Firmware Settings ...
Detecting snapshots ...
Info: Separate boot partition not detected 
Info: snapper detected, using config 'root'
Found snapshot: 2021-04-02 10:57:13 | @.snapshots/14/snapshot | single | writable copy of #5
Found snapshot: 2021-04-02 10:57:13 | @.snapshots/13/snapshot | single | rollback backup of #12
Found snapshot: 2021-04-02 10:29:57 | @.snapshots/12/snapshot | single | writable copy of #5
Found snapshot: 2021-04-02 10:29:57 | @.snapshots/11/snapshot | single | rollback backup of #10
Found snapshot: 2021-04-02 10:24:23 | @.snapshots/10/snapshot | single | writable copy of #5
Found snapshot: 2021-04-02 10:24:23 | @.snapshots/9/snapshot  | single | rollback backup of #8
Found snapshot: 2021-04-02 10:22:24 | @.snapshots/8/snapshot  | single | writable copy of #5
Found snapshot: 2021-04-02 10:22:24 | @.snapshots/7/snapshot  | single | rollback backup
Found snapshot: 2021-04-02 10:18:49 | @.snapshots/6/snapshot  | post   | micro wl-clipboard
Found snapshot: 2021-04-02 10:18:49 | @.snapshots/5/snapshot  | pre    | /usr/bin/pamac-daemon
Found snapshot: 2021-04-02 10:18:28 | @.snapshots/4/snapshot  | post   | firefox gnome-shell-extension-arcmenu gnome-shell-extension-dash-to-dock...
Found snapshot: 2021-04-02 10:18:25 | @.snapshots/3/snapshot  | pre    | /usr/bin/pamac-daemon
Found snapshot: 2021-04-02 09:46:00 | @.snapshots/2/snapshot  | post   | snap-pac-grub
Found snapshot: 2021-04-02 09:46:00 | @.snapshots/1/snapshot  | pre    | pacman -U --noconfirm --config /etc/pacman.conf -- /home/nightliss/.cach...
Found 14 snapshot(s)
Found memtest86+ image: /boot/memtest86+/memtest.bin
done
Installing for x86_64-efi platform.
grub-install: error: cannot find EFI directory.
```

after trial and error I realized that grub cannot find efi directory because it's not mounted.

outside chroot
```
mount | grep /boot/efi
/dev/sda1 on /boot/efi type vfat (rw,relatime,fmask=0077,dmask=0077.,codepage=437,iocharset=ascii,shortname=mixed,utf8,errors=remount-ro)
```
I don't see anything in the script which would mount boot partition.

I have followed the guide from the video link in the readme.

```
sudo parted -l
[sudo] password for nightliss: 
Model: ATA VBOX HARDDISK (scsi)
Disk /dev/sda: 21,5GB
Sector size (logical/physical): 512B/512B
Partition Table: gpt
Disk Flags: 

Number  Start   End     Size    File system     Name  Flags
 1      1049kB  538MB   537MB   fat32                 boot, esp
 2      538MB   2685MB  2147MB  linux-swap(v1)        swap
 3      2685MB  21,5GB  18,8GB  btrfs
```

Does the script assumes there is only root partition with boot or is this a bug?

**Comment by hrotkogabor** [comment]
Hi,

 indeed, these scripts are not prepared for UEFI, because I do not use that.
These days I will not have enough time to make the fix, so I welcome any patches to test.

thanx!

**Comment by hrotkogabor** [comment]
Fixed with 78d79d9
